packages/zowe-explorer/src/utils/CertificateWizard.ts

```diff
@@ -0,0 +1,127 @@
+import { Disposable, ExtensionContext, OpenDialogOptions, Uri } from "vscode";
+import * as nls from "vscode-nls";
+import { ZoweLogger } from "./LoggerUtils";
+
+// Set up localization
+nls.config({
+    messageFormat: nls.MessageFormat.bundle,
+    bundleFormat: nls.BundleFormat.standalone,
+})();
+const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+export type CertWizardOpts = {
+    cert?: string;
+    certKey?: string;
+    dialogOpts?: OpenDialogOptions;
+};
+
+class DeferredPromise<T> {
+    public promise: Promise<T>;
+    public resolve: (value: T | PromiseLike<T>) => void;
+    public reject: (reason?: any) => void;
+
+    public constructor() {
+        this.promise = new Promise<T>((resolve, reject) => {
+            this.resolve = resolve;
+            this.reject = reject;
+        });
+    }
+}
+
+const userDismissed = localize("zowe.certWizard.dismissed", "User dismissed the certificate wizard.");
+
+export class CertificateWizard extends WebView {
+    private onUpdateDisposable: Disposable;
+    private opts: CertWizardOpts;
+    public userSubmission: DeferredPromise<{
+        cert: string;
+        certKey: string;
+    }> = new DeferredPromise();
+
+    public constructor(context: ExtensionContext, opts: CertWizardOpts) {
+        super("Certificate Wizard", "certificate-wizard", context, (message: object) => this.onDidReceiveMessage(message));
+        this.opts = opts;
+        this.panel.onDidDispose(() => {
+            this.userSubmission.reject(userDismissed);
+        });
+    }
+
+    protected async onDidReceiveMessage(message: any): Promise<void> {
+        switch (message.command) {
+            case "promptCert":
+                {
+                    const tempCert = await Gui.showOpenDialog({
+                        title: "Enter the path to the certificate for authenticating the connection.",
+                        canSelectFiles: true,
+                        canSelectFolders: false,
+                        canSelectMany: false,
+                        defaultUri: this.opts.cert ? Uri.file(this.opts.cert) : undefined,
+                        filters: { "Certificate Files": ["cer", "crt", "pem"], "All Files": ["*"] },
+                        openLabel: "Select Certificate",
+                        ...(this.opts.dialogOpts ?? {}),
+                    });
+                    if (tempCert != null && tempCert[0] != null) {
+                        this.opts.cert = tempCert[0].fsPath;
+                        await this.panel.webview.postMessage({
+                            opts: {
+                                cert: tempCert[0].fsPath,
+                            },
+                        });
+                    }
+                }
+                break;
+            case "promptCertKey":
+                {
+                    const tempCertKey = await Gui.showOpenDialog({
+                        title: "Enter the path to the certificate key for authenticating the connection.",
```

Comment:
Should the titles be localized as well? 😋

Reply:
Yes, addressed in next commit 👍